auslib/web/admin/swagger/api.yaml

```diff
@@ -3402,6 +3402,232 @@ paths:
         '404':
           $ref: '#/responses/resourceNotFound'
 
+  /emergency_shutoff:
+    get:
+      summary: List of emergency shut off optionally filtered by product/channel.
+      description: >
+        Products/Channels available to shut off updates.
+      tags:
+        - Emergency Shutoff
+      operationId: auslib.web.common.emergency_shutoff.get
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/productParam'
+        - $ref: '#/parameters/channelQueryParam'
+      responses:
+        '200':
+          description: Emergency shut off list.
+          schema:
+            $ref: '#/definitions/EmergencyShutoffListGET'
+
+    post:
+      summary: Creates an emergency shut off.
+      description: Creates an emergency shut off for given product/channel.
+      tags:
+        - Emergency Shutoff
+      operationId: auslib.web.admin.emergency_shutoff.post
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: emergency_shutoff
+          in: body
+          description: Emergency shut off data.
+          schema:
+            allOf:
+              - $ref: '#/definitions/EmergencyShutOffModel'
+      responses:
+        '201':
+          description: Successfully created an emergency shut off.
+        '400':
+          $ref: '#/responses/invalidFormData'
+
+  /emergency_shutoff/{shutoff_id}:
+    get:
+      summary: Emergency shut off.
+      description: >
+        Get emergency shut off by Id.
+      tags:
+        - Emergency Shutoff
+      operationId: auslib.web.common.emergency_shutoff.get_by_id
+      produces:
+        - application/json
+      parameters:
+        - $ref: '#/parameters/shutoff_id'
+      responses:
+        '200':
+          description: Shut off information.
+          schema:
+            $ref: "#/definitions/EmergencyShutoffObject"
+          headers:
+            X-Data-Version:
+              description: data version of the shut off
+              type: integer
+              format: int32
+              minimum: 1
+        '404':
+          $ref: '#/responses/resourceNotFound'
+
+    put:
+      summary: Update an emergency shut off.
+      description: Update an emergency shut off. The product/channel can not be the same than an existing emergency shut off.
```

Comment:
Does this endpoint allow someone to change the product or channel of an existing emergency shutoff? If so, that's probably confusing. It would be better to have users delete the existing shutoff and create an entirely new one. The only "update" that should happen to an existing shutoff is changing whether or not updates are disabled or not.

Reply:
Sure, I was in doubt by include or not this update endpoint. Will be removed.